### DISCUSSION TITLE
Consider creating a "reference terminal" program

### DISCUSSION THREAD

**Discussion by certik** [Textualize/textual#2439] [Ideas]
It looks like almost all existing terminals have issues, such as:

* macOS default terminal: https://github.com/Textualize/textual/issues/2435
* macOS kitty: https://github.com/Textualize/textual/issues/2436

I can see two ways forward:

* Having a set of recommended terminals that work well, together with the exact configuration options and fonts and document how to set it up
* Create a "reference terminal" app. It doesn't need to work with all corner cases, which I think is very hard. But it would "just" work as long as the program running in the terminal is a Rich/Textual app. Since you would control both the underlying rendering and input framework and the terminal, I think this might be doable. In a way, that is probably the plan with the web application backend. But I like TUI, and having a reference terminal could push the state of the art. Also you could then support high resolution images in the terminal, using for example the protocol in kitty. By providing more "competition" in the terminal business, other terminals could catch up, similarly like it works with the web browsers.

**Comment by davep** [comment]
On macOS, at least, I've found both iTerm and kitty have worked really well for me. There are people who speak highly of alacritty too. While it's not my position to yay or nay this idea (in the end that's up to Will), I would suggest that if Textual is highlighting a shortcoming in an actively-developed/supported terminal application, it better serves its user-base and the FOSS community as a whole if that feedback goes back to those projects so everyone benefits.

**Reply by certik** [reply]
I use the default terminal, as do most of my users/collaborators on macOS. The default terminal does not work well and it is not supported/recommended it seems. Is there a place to report these bugs to it?

I think maybe the answer is that such "reference terminal" could be either Kitty or iTerm on macOS, and if something doesn't work there, it can be reported to them and ensure it gets fixed (either by submitting PRs or by the upstream developers). I just checked iTerm and that actually seems to work well with Textual.

**Comment by willmcgugan** [comment]
It's impractical for use to build a "reference terminal" right now. Even one just designed to run Textual. It's a lot of work.

On macOS, iTerm is the standard for us. Not the fastest, but it is mostly free of glitches. At least until we add image support.
